apps/web/src/components/transfer/TransferAsset.tsx

```diff
@@ -66,10 +73,51 @@ const TransferAsset = () => {
     useState<SupportedNetwork>(DEFAULT_DESTINATION)
   const [allowedTokens, setAllowedTokens] = useState([])
 
+  const [maxAmount, setMaxAmount] = useState(0)
+  const { balances } = useWalletBalance({
+    //, isLoading, fetchBalances
+    currentAccount,
+    currentChainId: selectedOrigin.prefix,
+    currentNetwork: selectedOrigin,
+  })
+
   const handleClose = () => {
     setCloseMenu(false)
   }
 
+  const calculateMaxAmountEVM = async () => {
+    //console.log('metamask balance ', selectedToken)
+    //console.log(balances, typeof(balances))
+    const am = balances
```

Comment:
let's not use abbreviation to save few bytes, we should have meaningfull naming convention, that explains the intent or/and comments.